src/query/functions-v2/src/scalars/variant.rs

```diff
@@ -43,4 +75,756 @@ pub fn register(registry: &mut FunctionRegistry) {
             Ok(())
         }),
     );
+
+    registry.register_passthrough_nullable_1_arg::<VariantType, VariantType, _, _>(
```

Comment:
`VaraintType` should be registered as the last resort, otherwise, numbers will be cast to variant at the call of `parse_json`.